components/UnifiedDocFeed/FeedMenu/FeedMenuTab.tsx

```diff
@@ -38,40 +41,46 @@ const FeedMenuTab = ({
         styles.tab,
         isSelected && styles.tabSelected
       )} typeFilter`}
-      onClick={() => {
-        if (isSelected) {
-          // Already handled by <Link>
-          return null;
-        }
-
-        if (nestedOptions.length > 0) {
-          if (isTagsMenuOpen) {
-            handleOpenTagsMenu(null);
-          } else {
-            handleOpenTagsMenu(tabObj.value);
-          }
-        }
-      }}
     >
       <Link href={url}>
-        <a className={css(styles.labelContainer)}>
+        <a
+          className={css(styles.labelContainer)}
+          onClick={(event) => {
+            const thisTabIsSelected = isSelected;
+            const thisTabHasTags = nestedOptions.length > 0;
+
+            if (thisTabIsSelected && thisTabHasTags) {
+              event.preventDefault();
+              if (isTagsMenuOpen) {
+                handleOpenTagsMenu(null);
+              } else {
+                handleOpenTagsMenu(tabObj.value);
+              }
```

Comment:
Just a preference to avoid nested if-s
```suggestion
                handleOpenTagsMenu(isTagsMenuOpen ? null : abObj.value);
```